### PR TITLE
Storages: Fix memory trace for UniversalPageId

### LINKED ISSUE #8835: Support thread-wise memory alloc/dealloc monitor
## Enhancement

Basicly https://github.com/tikv/tikv/issues/8235, but with some tricks to fix the two copies of jemalloc problem

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: close #8835

Problem Summary:
We don't trace when a UniversalPageId is move-constructed or move-assigned, so we may count a memory release twice

image

### What is changed and how it works?

```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp [hunk 1]
```diff
@@ -15,6 +15,44 @@
...
+UniversalPageId & UniversalPageId::operator=(UniversalPageId && other) noexcept
+{
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(size());
+    id = std::move(other.id);
+    return *this;
+}
+UniversalPageId & UniversalPageId::operator=(const UniversalPageId & other) noexcept
+{
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(size());
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_add(other.size());
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp:43]:
```suggestion
    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(id.size());
    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_add(other.id.size());
```

File: dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp [hunk 2]
```diff
@@ -15,6 +15,44 @@
...
+    id = std::move(other.id);
+}
+UniversalPageId::UniversalPageId(const UniversalPageId & other)
+{
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_add(other.size());
+    id = other.id;
+}
+UniversalPageId & UniversalPageId::operator=(UniversalPageId && other) noexcept
+{
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(size());
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp:36]:
```suggestion
    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(id.size());
```

File: dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp [hunk 3]
```diff
@@ -15,6 +15,44 @@
...
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(id.size());
+}
+
+UniversalPageId::UniversalPageId(UniversalPageId && other)
+{
+    id = std::move(other.id);
+}
+UniversalPageId::UniversalPageId(const UniversalPageId & other)
+{
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_add(other.size());
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp:31]:
```suggestion
    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_add(other.id.size());
```

File: dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp [hunk 4]
```diff
@@ -15,6 +15,44 @@
...
+namespace DB
+{
+UniversalPageId::~UniversalPageId()
+{
+    PS::PageStorageMemorySummary::uni_page_id_bytes.fetch_sub(id.size());
+}
+
+UniversalPageId::UniversalPageId(UniversalPageId && other)
+{
+    id = std::move(other.id);
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Page/V3/Universal/UniversalPageId.cpp:27]:
```suggestion
    // PS::PageStorageMemorySummary::uni_page_id_bytes has been set when `other` created
    id = std::move(other.id);
```

**Review by Lloyd-Pottiger — APPROVED** [review]

**Review by JaySon-Huang — APPROVED** [review]

**Comment by Lloyd-Pottiger** [comment]
/hold

TODO: fix tidy

**Pull request merged**